fun/management/commands/export_course_anonymized_demographics.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+from django.contrib.auth.models import User
+from django.core.management.base import BaseCommand, CommandError
+from django.conf import settings
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys import InvalidKeyError
+from optparse import make_option
+from student.models import CourseEnrollment
+from student.models import UserProfile
+from xmodule.modulestore.django import modulestore
+import hashlib
+import hmac
+import sys
+import logging
+
+logger = logging.getLogger(__name__)
+#pour ques les anonymized_username soient les meme que dans les tracking log, il faudra utiliser la même ANONYMIZATION_KEY que pour ceux-ci
+
+class Command(BaseCommand):
+    help = """Gather demographics from a course_id, anonymize users and prints the result as a JSON """
+    option_list = BaseCommand.option_list + ( make_option('--course_id', action='store', dest='COURSE_ID', type='string' ) , )
+
+    def get_courses(self):
+        store = modulestore()
+        return store.get_courses()
+
+    def handle(self, *args, **options):
+        if not all([options['COURSE_ID']]):
+            raise CommandError('--course_id mandatory')
+        try:
+            course_key = CourseKey.from_string(options['COURSE_ID'])
+            if not modulestore().get_course(course_key, depth=2):
+                raise Exception
+        except InvalidKeyError:
+            print("Course id {} could not be parsed as a CourseKey;".format(options['COURSE_ID']))
+        except :
+            print  >> sys.stderr, "\n Course {} not found.".format(options['COURSE_ID'])
+            sys.exit(-1)
```

Comment:
En fait, il serait peut-être plus judicieux de terminer cette commande par : 

    raise CommandError("\n Course {} not found.".format(options['COURSE_ID']))

Ca permettrait de ne pas faire quitter un process qui appelle cette fonction, comme les tests unitaires ou (qui sait ?) gunicorn.